### PR TITLE
Query node input schema

### PR DESCRIPTION
**Pull request by ondratra**
Input schema for query node reflecting the state of the new content directory.

### CODE REVIEW

**Comment by bedeho** [comment]
Am I supposed to review this? seems a bit premature, the same problems are still here? @mnaamani ?

**Comment by mnaamani** [comment]
> Am I supposed to review this? seems a bit premature, the same problems are still here? @mnaamani ?

Not yet, It hasn't been updated yet following my discussion with Andy. I just tagged you in it to stay updated with changes.

**Comment by mnaamani** [comment]
@ondratra I would avoid making any changes to code in the runtime in this PR (even though as far as I can see they are just code comments). It would be better to collect any changes specific for runtime separately.

**Comment by ondratra** [comment]
Only input schema related changes are left in this PR. Other changes previously included in this PR were removed and will be part of other PRs that will come in the future.

File: query-node/schema.graphql [hunk 1]
```diff
@@ -12,32 +12,127 @@ type Block @entity {
...
   "Short text chosen by member to share information about themselves"
-  about: String
-
-  "Blocknumber when member was registered"
-  registeredAtBlock: Int!
+  about: String!
 
   "Member's controller account id"
-  controllerAccount: Bytes!
+  controllerAccount: String!
```
Comment (mnaamani) [inline query-node/schema.graphql:41]: Why the change from Bytes to String? (I assume Bytes would have represented the 32Byte raw account id and the string will be the encoded address?)

File: query-node/schema.graphql [hunk 2]
```diff
@@ -12,32 +12,127 @@ type Block @entity {
...
   "MemberId: runtime identifier for a user"
   id: ID!
 
   "The unique handle chosen by member"
-  handle: String @unique @fulltext(query: "membersByHandle")
+  handle: String! @unique @fulltext(query: "membersByHandle")
 
   "A Url to member's Avatar image"
-  avatarUri: String
+  avatarUri: String!
```
Comment (mnaamani) [inline query-node/schema.graphql]: avatar should be nullable because the runtime allows that.

File: query-node/schema.graphql [hunk 3]
```diff
@@ -12,32 +12,127 @@ type Block @entity {
...
 
   "Short text chosen by member to share information about themselves"
-  about: String
-
-  "Blocknumber when member was registered"
-  registeredAtBlock: Int!
+  about: String!
```
Comment (mnaamani) [inline query-node/schema.graphql]: The about field should also be nullable because the runtime allows it. Can we still have an empty string if we use `String!` ?

File: query-node/schema.graphql [hunk 4]
```diff
@@ -12,32 +12,127 @@ type Block @entity {
...
+  entry: MembershipEntryMethod!
+
+  "The type of subscription the member has purchased if any."
+  subscription: BigInt
+}
+
+# TODO: add description to fields
+# TODO: revise property requirement for the new and existing fields derived from content-metadata-protobuf
+
+type Account @entity {
```
Comment (mnaamani) [inline query-node/schema.graphql]: This type isn't used anywhere. What is it for?

File: query-node/schema.graphql [hunk 5]
```diff
@@ -12,32 +12,127 @@ type Block @entity {
...
+
+  firstName: String
+
+  middleName: String
+
+  lastName: String
+
+  about: String
+
+  coverPhotoUrl: String
```
Comment (mnaamani) [inline query-node/schema.graphql]: This is an asset which may be a url or content in the storage system identified by its `ContentId`.
So both the name of the field and the type is not correct.

File: query-node/schema.graphql [hunk 6]
```diff
@@ -12,32 +12,127 @@ type Block @entity {
...
+
+  avatarPhotoUrl: String
```
Comment (mnaamani) [inline query-node/schema.graphql]: Same as `coverPhoto`

File: query-node/schema.graphql [hunk 7]
```diff
@@ -12,32 +12,127 @@ type Block @entity {
...
+
+  people: [Person]
+
+  seasons: [Season]
+}
+
+type Asset @entity {
+  id: ID!
+
+  url: String!
```
Comment (mnaamani) [inline query-node/schema.graphql]: An asset needs to represent the possible types, either a Url or content in the storage system. And in the latter case the upload status. see: https://github.com/Joystream/joystream/issues/2033

File: query-node/schema.graphql [hunk 8]
```diff
@@ -98,9 +195,6 @@ type Channel @entity {
   "Flag signaling whether a channel is public."
   isPublic: Boolean!
 
-  "Flag signaling whether a channel is curated/verified."
-  isCurated: Boolean!
```
Comment (mnaamani) [inline query-node/schema.graphql:102]: replaced with `is_censored`

File: query-node/schema.graphql [hunk 9]
```diff
@@ -83,6 +178,8 @@ type Channel @entity {
   # or a member. We are not handling events related to curator group so we will not set this field
   # owner: Member!
 
+  category: ChannelCategory
+
```
Comment (mnaamani) [inline query-node/schema.graphql:212]: handle is renamed to `title` and is not unique

File: query-node/schema.graphql [hunk 10]
```diff
@@ -203,6 +297,24 @@ type HttpMediaLocationEntity @entity {
   happenedIn: Block!
 }
 
+type MediaType @entity {
```
Comment (mnaamani) [inline query-node/schema.graphql]: I don't think we need to populate the query node with media types or languages

File: query-node/schema.graphql [hunk 11]
```diff
@@ -226,6 +338,8 @@ type VideoMedia @entity {
 
   locationEntity: MediaLocationEntity
 
+  publishedBeforeJoystream: PublishedBeforeJoystream
+
```
Comment (mnaamani) [inline query-node/schema.graphql]: I think the concept of two separate types `VideoMedia` and `Video` was relic of old content directory design. There is only one concept of a `Video` in the content directory. Is there reason to have these two types still?

**Review by mnaamani — CHANGES_REQUESTED** [review]
I'm not super familiar with the syntax here but there seems to be lots of gaps and in some cases some misunderstanding of some of the key ideas that need to be captured in this schema model. Specifically around assets and storage.

File: query-node/schema.graphql (hunk 8 above)
Comment (ondratra) [reply query-node/schema.graphql:102]: Done

File: query-node/schema.graphql (hunk 1 above)
Comment (ondratra) [reply query-node/schema.graphql:41]: I copied the membership from https://github.com/Joystream/joystream/pull/2105 where this change was present. I guess the point was to have it in some readable format for easy conversion to JSON, but I will ask Metin to double-check.

File: query-node/schema.graphql (hunk 2 above)
Comment (ondratra) [reply query-node/schema.graphql]: Done

File: query-node/schema.graphql (hunk 4 above)
Comment (ondratra) [reply query-node/schema.graphql]: I forgot to remove this one. Initially, I thought an entity for an account would be handy there, but since it has no properties, it is redundant. Done.

File: query-node/schema.graphql (hunk 5 above)
Comment (ondratra) [reply query-node/schema.graphql]: Done

File: query-node/schema.graphql (hunk 7 above)
Comment (ondratra) [reply query-node/schema.graphql]: Asset upgraded. Check if it's ok, please.

File: query-node/schema.graphql (hunk 9 above)
Comment (ondratra) [reply query-node/schema.graphql:212]: Done

File: query-node/schema.graphql (hunk 10 above)
Comment (ondratra) [reply query-node/schema.graphql]: I removed the media type. Should the language be completely removed as well? Has it been removed from metadata protobuf recently? Just double-checking.

File: query-node/schema.graphql (hunk 11 above)
Comment (ondratra) [reply query-node/schema.graphql]: Great, that simplifies the video structure in my eyes. Please check that the new structure is ok (and that assets are used correctly).

File: query-node/schema.graphql (hunk 6 above)
Comment (ondratra) [reply query-node/schema.graphql]: Done

File: query-node/schema.graphql (hunk 3 above)
Comment (ondratra) [reply query-node/schema.graphql]: Done. The exclamation mark means non-nullable - it doesn't prevent empty strings. I'll discuss it with Atlas team if it is desirable to ensure non-empty strings. If that's the case, we should probably be consistent about it everywhere a `String` is used.

File: query-node/schema.graphql (hunk 1 above)
Comment (ondratra) [reply query-node/schema.graphql:41]: The String is really used just for convenient work with the data in sandboxes, while building apps, etc.

File: query-node/schema.graphql (hunk 3 above)
Comment (ondratra) [reply query-node/schema.graphql]: Confirmed from Atlas team that non-nullable string is the desired state and empty strings are ok there.

**Review by bedeho — COMMENTED** [review]
Overall this looks good and correct.

I want to warn that all of the deletion stuff, both for channels and channel assets, is not really safe. The number of assets and objects could be huge here, and even if we put a limit, it would not be cost effective weight, because of so many database reads/writes, so this approach will need to be changed, probably by inlining channel objects, not in separate maps. Same goes for storage assets, they will be managed differently in the future.

**Comment by mnaamani** [comment]
@bedeho 
> I want to warn that all of the deletion stuff, both for channels and channel assets, is not really safe.

Are you referring to just the query node or in runtime as well. I'm inclined to remove support for deleting channels in runtime after some discussion with @bwhm https://github.com/Joystream/joystream/issues/2165#issuecomment-782098356 https://github.com/Joystream/joystream/pull/2158#issuecomment-781960074

However deleting assets just still be supported otherwise when a channel reaches its quota limit in storage and owner wants to remove some old videos lets say to make room for new content they will not be able to.

**Comment by metmirr** [comment]
I have two suggestions:

1. I suggest we use `String` for accounts instead of `Bytes` which is Hex string in the graphql api. I can tell that accounts are human readable on all substrate explorers. Example: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY' is will look like `0x307864343335393363373135666464333163363131343161626430346139396664363832326338353538383534636364653339613536383465376135366461323764`
2. If I am not mistaken Atlas team requested to send `DateTime` for `publishedBeforeJoystream` field and Hydra support `DateTime` scalar type. Substrate's timestamp(`Moment`) is compatible with javascript. So mapping author can create `Date` objects without any problem.

**Comment by bedeho** [comment]
> Overall this looks good and correct.
> 
> I want to warn that all of the deletion stuff, both for channels and channel assets, is not really safe. The number of assets and objects could be huge here, and even if we put a limit, it would not be cost effective weight, because of so many database reads/writes, so this approach will need to be changed, probably by inlining channel objects, not in separate maps. Same goes for storage assets, they will be managed differently in the future.

I have no clue how this happened, probably me getting confused, but this reply was meant for this PR: https://github.com/Joystream/joystream/pull/2158

Terribly sorry @mnaamani @ondratra .

File: query-node/schema.graphql [hunk 12]
```diff
@@ -12,30 +12,142 @@ type Block @entity {
...
+# TODO: add description to fields
+# TODO: revise property requirement for the new and existing fields derived from content-metadata-protobuf
+
+type ChannelCategory @entity {
+  id: ID!
+
+  name: String
+}
+
+type Person @entity {
```
Comment (mnaamani) [inline query-node/schema.graphql]: Lets not include Person, Series, Season.. anything that will not be part of Sumer release

File: query-node/schema.graphql [hunk 13]
```diff
@@ -12,30 +12,142 @@ type Block @entity {
...
+  description: String
+
+  coverPhoto: String
+
+  people: [Person!]
+
+  seasons: [Season!]
+}
+
+union Asset = AssetUrl | AssetStorage
```
Comment (mnaamani) [inline query-node/schema.graphql]: An Asset may also not be provided at all so it can be null

File: query-node/schema.graphql [hunk 14]
```diff
@@ -12,30 +12,142 @@ type Block @entity {
...
+
+type AssetUrl @variant {
+  id: ID!
+
+  "The http url pointing to the media"
+  url: String!
+
+  "The port to use when connecting to the http url (defaults to 80)"
+  port: Int
```
Comment (mnaamani) [inline query-node/schema.graphql]: I don't see a need to separate url and port, just use a single string as will be provided by the metadata

File: query-node/schema.graphql [hunk 15]
```diff
@@ -12,30 +12,142 @@ type Block @entity {
...
+}
+
+enum AssetStorageUploadStatus {
+  INCOMPLETE_FIRST_UPLOAD
+  INCOMPLETE_LATER_UPLOAD
+  CONFIRMED_UPLOADED
```
Comment (mnaamani) [inline query-node/schema.graphql]: The discussion #2033 had other recommendations that were an improvement on this initial design.

File: query-node/schema.graphql [hunk 16]
```diff
@@ -83,23 +195,25 @@ type Channel @entity {
...
   "The title of the Channel"
-  handle: String! @fulltext(query: "search")
+  title: String! @fulltext(query: "search")
 
   "The description of a Channel"
   description: String!
 
   "Url for Channel's cover (background) photo. Recommended ratio: 16:9."
-  coverPhotoUrl: String
+  coverPhotoUrl: Asset
```
Comment (mnaamani) [inline query-node/schema.graphql]: Either the `Asset` should be `union Asset = AssetUrl | AssetStorage | Null` or this filed should be nullable.
the Url should be removed from the field name because it may not be a url.

Comment (mnaamani) [reply query-node/schema.graphql]: All the fields that come from protobuf metadata should be nullable because they are optional

File: query-node/schema.graphql [hunk 17]
```diff
@@ -269,37 +322,40 @@ type Video @entity {
...
+  "Flag signaling whether a video is censored."
+  isCensored: Boolean!
 
   "Whether the Video contains explicit material."
   isExplicit: Boolean!
 
   license: LicenseEntity!
 
+  "Encoding of the video media object"
+  encoding: VideoMediaEncoding!
```
Comment (mnaamani) [inline query-node/schema.graphql]: VideoMediaEncoding is not correct, its the old content directory schema entity. In new design it comes from the protobuf metadata. And I don't think we will have 'pre-populated' encoding entities in the query-node database. Or should we?

File: query-node/schema.graphql (hunk 12 above)
Comment (ondratra) [reply query-node/schema.graphql]: Person, Series, Season, and Playlist were removed.

File: query-node/schema.graphql (hunk 13 above)
Comment (ondratra) [reply query-node/schema.graphql]: Asset is nullable already. It can't be seen in the asset definition itself, but everywhere it is used, it is nullable (`myProperty: Asset` - without exclamation mark that marks non-nullable properties).

File: query-node/schema.graphql (hunk 14 above)
Comment (ondratra) [reply query-node/schema.graphql]: Port removed.

File: query-node/schema.graphql (hunk 16 above)
Comment (ondratra) [reply query-node/schema.graphql]: As stated above: Asset is nullable already. It can't be seen in the asset definition itself, but everywhere it is used, it is nullable (`myProperty: Asset` - without exclamation mark that marks non-nullable properties).

Comment (ondratra) [reply query-node/schema.graphql]: I revised all content coming from protobuf and made it nullable in a couple of places where it was previously non-nullable.

File: query-node/schema.graphql (hunk 10 above)
Comment (ondratra) [reply query-node/schema.graphql]: Language stays and they will be identified via ISO_639-1 string.

File: query-node/schema.graphql (hunk 17 above)
Comment (ondratra) [reply query-node/schema.graphql]: VideoMediaEncoding entity has been removed and relevant properties from protobuf have been integrated into Video entity.

**Comment by ondratra** [comment]
The changes requested in the last review have been reflected in the input schema.

Other changes that have been discussed outside of this GitHub issue:
- `PublishedBeforeJoystream` integrated into Video
- language representation as ISO string
- timestamps converted to `DateTime`
- storage asset upload status improved

File: query-node/schema.graphql [hunk 18]
```diff
@@ -8,36 +13,149 @@ type Block @entity {
...
+
+  "Storage provider as liaison judgment"
+  liaisonJudgement: LiaisonJudgement!
+
+  "IPFS content id"
+  ipfsContentId: String!
+}
+
+"Owner type for storage object"
+enum AssetOwner {
```
Comment (mnaamani) [inline query-node/schema.graphql]: Can enum variants enclose values? Because variant `Member` is not enough to capture which member id is the owner for example

File: query-node/schema.graphql [hunk 19]
```diff
@@ -83,23 +201,25 @@ type Channel @entity {
...
-  description: String!
+  description: String
 
   "Url for Channel's cover (background) photo. Recommended ratio: 16:9."
-  coverPhotoUrl: String
+  coverPhotoUrl: Asset
 
   "Channel's avatar photo."
-  avatarPhotoUrl: String
+  avatarPhotoUrl: Asset
```
Comment (mnaamani) [inline query-node/schema.graphql]: the field name should not have `Url` in it

File: query-node/schema.graphql [hunk 20]
```diff
@@ -237,45 +281,60 @@ type Video @entity {
   channel: Channel!
 
   "Reference to a video category"
-  category: Category!
+  category: Category
```
Comment (mnaamani) [inline query-node/schema.graphql]: rename to `VideoCategory`

`ChannelCategory` is very simple:
```
"Category of media channel"
type ChannelCategory @entity {
  id: ID!

  name: String
}
```

and the `Category` you are using here as the VideoCategory has name and description, but profobuf message only has `name`. Can you make both types more consistent.

File: query-node/schema.graphql [hunk 21]
```diff
@@ -237,45 +281,60 @@ type Video @entity {
...
   "The description of the Video"
-  description: String!
+  description: String
 
   "Video duration in seconds"
-  duration: Int!
+  duration: Int
 
   "Video's skippable intro duration in seconds"
   skippableIntroDuration: Int
```
Comment (mnaamani) [inline query-node/schema.graphql]: I think `skippableIntroDuration` is missing from protobuf definition. Is this required in Atlas?

File: query-node/schema.graphql [hunk 22]
```diff
@@ -237,45 +281,60 @@ type Video @entity {
...
 
   "Video thumbnail url (recommended ratio: 16:9)"
-  thumbnailUrl: String!
+  thumbnailUrl: String
```
Comment (mnaamani) [inline query-node/schema.graphql]: This is an Asset not a string url

File: query-node/schema.graphql [hunk 23]
```diff
@@ -237,45 +281,60 @@ type Video @entity {
...
 
   "Video's main langauge"
   language: Language
 
-  "Reference to VideoMedia"
-  media: VideoMedia!
```
Comment (mnaamani) [inline query-node/schema.graphql:261]: We need an Asset in its place for the actual video

File: query-node/schema.graphql [hunk 24]
```diff
@@ -237,45 +281,60 @@ type Video @entity {
...
-  "Video curation status set by the Curator"
-  isCurated: Boolean!
+  "Flag signaling whether a video is censored."
+  isCensored: Boolean!
 
   "Whether the Video contains explicit material."
-  isExplicit: Boolean!
+  isExplicit: Boolean
+
+  license: LicenseEntity
```
Comment (mnaamani) [inline query-node/schema.graphql]: revise the LicenseEntity type it doesn't match with our design anymore: https://github.com/Joystream/joystream/blob/sumer/content-metadata-protobuf/doc/index.md#license

**Review by mnaamani — CHANGES_REQUESTED** [review]
Left some more comments

File: query-node/schema.graphql (hunk 21 above)
Comment (kdembler) [reply query-node/schema.graphql]: No, it's not being used currently. I think it was intended as a future feature

File: query-node/schema.graphql [hunk 25]
```diff
@@ -237,45 +281,60 @@ type Video @entity {
...
+
+  "Encoding of the video media object"
+  codecName: String
```
Comment (kdembler) [inline query-node/schema.graphql]: Any specific reason to drop the VideoMedia and add the fields directly here? I personally liked that separation of video metadata and video file metadata

File: query-node/schema.graphql [hunk 26]
```diff
@@ -83,23 +201,25 @@ type Channel @entity {
...
-  handle: String! @fulltext(query: "search")
+  title: String @fulltext(query: "search")
 
   "The description of a Channel"
-  description: String!
+  description: String
 
   "Url for Channel's cover (background) photo. Recommended ratio: 16:9."
-  coverPhotoUrl: String
+  coverPhotoUrl: Asset
```
Comment (kdembler) [inline query-node/schema.graphql]: We will need to use one common naming for the assets, `Url` suffix will not work anymore. I guess it could be just `coverPhoto`?

File: query-node/schema.graphql [hunk 27]
```diff
@@ -8,36 +13,149 @@ type Block @entity {
...
+  size: BigInt!
+
+  "Storage provider id of the liaison"
+  liaisonId: BigInt!
+
+  "Storage provider as liaison judgment"
+  liaisonJudgement: LiaisonJudgement!
+
+  "IPFS content id"
+  ipfsContentId: String!
```
Comment (kdembler) [inline query-node/schema.graphql:152]: Just to make sure, this is the old `dataObjectId`?

File: query-node/schema.graphql (hunk 26 above)
Comment (mnaamani) [reply query-node/schema.graphql]: Yes, I made the same point on another field also (thumbnailUrl -> thumbnail)

File: query-node/schema.graphql (hunk 19 above)
Comment (ondratra) [reply query-node/schema.graphql]: `Url` suffix removed

File: query-node/schema.graphql (hunk 20 above)
Comment (ondratra) [reply query-node/schema.graphql]: Types structure unified (`ChannelCategory` now looks like `VideoCategory`).

File: query-node/schema.graphql (hunk 21 above)
Comment (ondratra) [reply query-node/schema.graphql]: The field has been removed.

File: query-node/schema.graphql (hunk 22 above)
Comment (ondratra) [reply query-node/schema.graphql]: `Url` suffix removed

File: query-node/schema.graphql (hunk 23 above)
Comment (ondratra) [reply query-node/schema.graphql:261]: Done

File: query-node/schema.graphql (hunk 24 above)
Comment (ondratra) [reply query-node/schema.graphql]: `License` has been reworked according to metadata.

File: query-node/schema.graphql (hunk 26 above)
Comment (ondratra) [reply query-node/schema.graphql]: Done

File: query-node/schema.graphql (hunk 27 above)
Comment (ondratra) [reply query-node/schema.graphql:152]: I assume so since it's only `id` in `storage`'s `DataObjectInternal`.

File: query-node/schema.graphql (hunk 18 above)
Comment (ondratra) [reply query-node/schema.graphql]: I think they can't. Rework `AssetOwner` into a union of variants that can hold relevant ids.

File: query-node/schema.graphql (hunk 25 above)
Comment (ondratra) [reply query-node/schema.graphql]: We've discussed this with Mokhtar on the call, and if I understood it correctly, the reason was simply not to have 1:1 relations anymore, but maybe I'm mistaken. What do you think @mnaamani?

Comment (mnaamani) [reply query-node/schema.graphql]: My bad, I had asked Andy to drop the separate entities because I assumed this model was chosen to match explicitly the separate entities in the old content directory. In new content directory there isn't this explicit separation so we can revert if it makes more sense. - sorry @ondratra

File: query-node/schema.graphql (hunk 27 above)
Comment (mnaamani) [reply query-node/schema.graphql:152]: IPFS content id is the IPFS hash of the content. That is different than the data object id (also referred to as ContentId)

Comment (mnaamani) [reply query-node/schema.graphql:152]: Maybe rename it to avoid this confusion ?

**Comment by bedeho** [comment]
I only looked representation of `DataObject`, but I don't think this is quite right yet? It has some of the elements from this discussions 

https://github.com/Joystream/joystream/issues/2033

but it also has lots of stuff which is redundant? Perhaps I have not kept up with some discussion on this?

File: query-node/schema.graphql (hunk 27 above)
Comment (kdembler) [reply query-node/schema.graphql:152]: If this isn't the `dataObjectId`, then isn't it missing from `AssetDataObject`?

**Comment by bedeho** [comment]
> I only looked representation of `DataObject`, but I don't think this is quite right yet? It has some of the elements from this discussions
> 
> #2033
> 
> but it also has lots of stuff which is redundant? Perhaps I have not kept up with some discussion on this?

Disregard this comment, was plain wrong.

File: query-node/schema.graphql (hunk 25 above)
Comment (kdembler) [reply query-node/schema.graphql]: From Atlas side, it doesn't make that much of a difference, I think it's just better DX separated. If we don't care, let's keep the old version

File: query-node/schema.graphql (hunk 27 above)
Comment (ondratra) [reply query-node/schema.graphql:152]: The discussed id was really missing. It is now called `joystreamContentId` to distinguish it from IPFS id.

File: query-node/schema.graphql (hunk 25 above)
Comment (ondratra) [reply query-node/schema.graphql]: `VideoMedia` was reintroduced, but under the name `VideoMetadata`.

**Review by kdembler — APPROVED** [review]
LGTM, as far as I can tell seems to suit all the Atlas needs :+1:

File: query-node/schema.graphql (hunk 25 above)
Comment (kdembler) [reply query-node/schema.graphql]: Just a thought: maybe we should call this `VideoMediaMetadata` or something else? Generally, I'd expect the title for example to be `VideoMetadata` as well. I think the current name may not properly indicate that we're talking about the media

**Review by mnaamani — CHANGES_REQUESTED** [review]
We are almost there so final points:

- There are some old types that can be removed:
```
ClassEntity
NextEntityId
KnownLicenseEntity
UserDefinedLicenseEntity
```
- The Membership type is still using `Bytes` for account and I think we decided a String would be best.
- The Channel type is missing the optional rewardAccount field


There seems to be perhaps some typeo somewhere in the file because syntax highlighting appears to be broken when viewing the whole file.

**Comment by ondratra** [comment]
Old entities have been removed. Accounts are represented as strings now, and the channel now has a reward account. Multiple categories for videos and channels are now supported (reflects changes in https://github.com/Joystream/joystream/pull/2214).

Syntax highlighting in GitHub seems to be broken for graphql. I've tried to make some changes in a testing branch, but the block quotes highlight doesn't work properly and brokes the rest of the file.

File: query-node/schema.graphql (hunk 25 above)
Comment (kdembler) [reply query-node/schema.graphql]: @ondratra ping

Comment (ondratra) [reply query-node/schema.graphql]: Sorry, I've missed the last reply. The rename does make sense since other video-related properties can also be considered metadata, and this change better distinguishes them. The name has been changed, and the entity is called `VideoMediaMetadata` now.

File: query-node/schema.graphql (hunk 15 above)
Comment (ondratra) [reply query-node/schema.graphql]: The changes were made accordingly to the mentioned discussion.

**Comment by mnaamani** [comment]
> Old entities have been removed. Accounts are represented as strings now, and the channel now has a reward account. Multiple categories for videos and channels are now supported (reflects changes in #2214).

Well I did say "If we merge" :)  .. so please revert changes for multiple categories.

Otherwise this looks good to merge now.

**Comment by ondratra** [comment]
Multiple categories reverted.

**Review by mnaamani — APPROVED** [review]
